system/Traits/ConditionalTrait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Traits;
+
+trait ConditionalTrait
+{
+    /**
+     * Only runs the query when $condition evaluates to true
+     *
+     * @param array|bool|float|int|object|resource|string|null $condition
```

Comment:
```suggestion
     * @template TWhen of mixed
     *
     * @phpstan-param TWhen                        $condition
     * @phpstan-param callable(self, TWhen): mixed $callback
     * @phpstan-param (callable(self): mixed)|null $defaultCallback
```

Reply:
It seems this is not yet committed.

Reply:
He applied it separately:
https://github.com/codeigniter4/CodeIgniter4/commit/b0a1bea2d805d8aec332f9a1e9c29a991264188b